### DISCUSSION TITLE
Reworked chart view

### DISCUSSION THREAD

**Discussion by alexanderameye** [afadil/wealthfolio#68] [Ideas]
Currently, the chart view is a bit lacking and has some issues.

- always green, even if a loss was made
- no axis labels or information

<img width="926" alt="image" src="https://github.com/user-attachments/assets/ed4719b3-41c9-4bf1-8220-8449d14e96b6">

I would like to improve it a bit (still a proof of concept!)

- add Y axis and X axis data
- green or red based on periods of net positive/negative
- horizontal grid lines

<img width="926" alt="image" src="https://github.com/user-attachments/assets/eb54530c-46da-4c1a-a604-17823598ff47">

Any other things you'd like to see added?

Some more ideas I had
- Change X axis so the dates it displays is based on selected range: for example selecting 1M shows a tick every week, selecting 1Y shows a tick every 3 months etc.
- Show dividends as dots on the chart if there are any, hovering these would show a tooltip about the dividend

The main thing I'm not too sure about is this overview window.

<img width="1111" alt="image" src="https://github.com/user-attachments/assets/6dc06c47-5514-4c31-a564-c5dba9782786">

If the chart is red and green, the visual style of this screen changes since the gradient can't flow over to the bottom of the page.
I do like that the chart can be both red and green to clearly show net positive/negative, but maybe on this overview page, it should just show either red or green completely, based on whether you are positive/negative for the whole selected period. Then it could still flow over in to the bottom, and either be red or green. It makes sense that the chart on the overview page has a bit less details so I think this would be a good compromise.

**Comment by alexanderameye** [comment]
Updated date formatting + added min/max indicators.

<img width="924" alt="image" src="https://github.com/user-attachments/assets/9b39035e-c470-41a7-bdac-7eda98a3c7e4">

**Reply by nathanrdodson** [reply]
This looks great - my only suggestion was going to be to alter the date formatting, which you did wonderfully.

**Comment by afadil** [comment]
Hi @alexanderameye, thank you for your contribution to Wealthfolio.

The app’s focus is on tracking long-term investments and wealth building. My goal is to keep it simple with minimal features.
The chart on the homepage is meant to be a discreet background, providing users with a high-level overview of portfolio trends. Users can hover over specific points in time to see the value, but we don’t want to display axes or additional information on this page.
Also, the app doesn’t manage debt at the moment, so we wouldn’t have any negative values in the charts.

The chart for a specific symbol (when you click on a holding line) can be improved. However, the red should be relative to a specific starting price within a period, as stock prices can't be negative.

**Reply by alexanderameye** [reply]
Hey, I understand that the chart on the homepage is meant to be just as a background. I think it's fine if that stays as it is.

However, I do think axes and additional info can make sense for the detail view (the screenshots above are for the detail view)? Personally as a user I would like to see some values on the axes to get some info without having to hover over the chart.

About the negative values: I didn't have debt in mind, but if you can track stock purchases in the app, for sure your investment could have a net negative return right? If I invested 1K USD and the current value of my portfolio is 500 USD, I think it would be neat to see that visually.

If the app does not track debt, then the overview will indeed always be green since it is always > 0, but individual investments could still have a net negative return, especially stocks.

I think overall a balance can be found in keeping the app looking minimal and discreet, while still providing useful data where needed. In my personal opinion I have little use in seeing little charts and bars without values or percentages next to them. 

Having a roadmap would be great to kind of see the vision of this app so we can contribute accordingly.

Example: if you import a portfolio that has a net loss currently

<img width="871" alt="image" src="https://github.com/user-attachments/assets/3d66f385-42ac-45b4-a883-79520669d0ea">

**Reply by afadil** [reply]
I think your chart could be a nice enhancement for the holding symbol page.
We can use the average stock cost as reference so bellow this price the graph become red.
I posted a roadmap draft in a separate discussion #70 

PS: the main page chart track the value of the holding and not the performance. Shouldn’t be negative if you add all the activities including deposits, withdrawals, buy, sell…

**Reply by alexanderameye** [reply]
> I think your chart could be a nice enhancement for the holding symbol page. We can use the average stock cost as reference so bellow this price the graph become red. I posted a roadmap draft in a separate discussion #70
> 
> PS: the main page chart track the value of the holding and not the performance. Shouldn’t be negative if you add all the activities including deposits, withdrawals, buy, sell…

Alright when I get the time I'll see if can apply these changes to the holding symbol page.
